src/Dialect/ONNX/Transforms/Recompose.cpp

```diff
@@ -340,6 +340,185 @@ struct RecomposeLayerNormFromMulPattern : public OpRewritePattern<ONNXMulOp> {
+
+    // Get the number of split outputs
+    int64_t numSplits = splitOp.getNumResults();
+    if (numSplits != static_cast<int64_t>(convOps.size())) {
+      return failure();
+    }
+
+    // Ensure all ConvOps have the same kernel, stride, dilation, and padding
+    for (size_t i = 1; i < convOps.size(); ++i) {
+      if (convOps[i].getKernelShape() != firstConv.getKernelShape() ||
+          convOps[i].getStrides() != firstConv.getStrides() ||
+          convOps[i].getDilations() != firstConv.getDilations() ||
+          convOps[i].getPads() != firstConv.getPads()) {
+        return failure();
+      }
+    }
+
+    // Create correct IntegerAttrs
+    IntegerAttr axis0 = rewriter.getI64IntegerAttr(0);
+    IntegerType si64Type = rewriter.getIntegerType(64, /*isSigned=*/true);
+    IntegerAttr groupAttrVal =
+        IntegerAttr::get(si64Type, static_cast<int64_t>(numSplits));
+
+    // **Concatenating Conv Weights Correctly**
+    SmallVector<Value, 2> weightTensors;
+    int64_t total_C_out = 0;
+    for (auto conv : convOps) {
+      weightTensors.push_back(conv.getW());
+      auto wType = mlir::dyn_cast<RankedTensorType>(conv.getW().getType());
+      total_C_out += wType.getDimSize(0); // Summing output channels
+    }
+
+    // Compute correct concatenated shape
+    Type newWeightType = RankedTensorType::get(
+        {total_C_out, weightType.getDimSize(1), weightType.getDimSize(2),
+            weightType.getDimSize(3)},
+        weightType.getElementType());
+
+    // Create new concatenated weight tensor (concatenating along axis=0)
+    Location weightLoc =
+        firstConv.getW().getLoc(); // Get location from the first Conv weight
+    axis0 = IntegerAttr::get(si64Type, 0);
+
+    Value concatenatedWeight = rewriter.create<ONNXConcatOp>(
+        weightLoc, newWeightType, weightTensors, axis0);
+
+    // **Concatenating Bias Correctly**
+    SmallVector<Value, 2> biasTensors;
+    bool hasBias = llvm::all_of(convOps, [](ONNXConvOp conv) {
+      return conv.getB() && !mlir::isa<NoneType>(conv.getB().getType());
+    });
+
+    Value concatenatedBias = Value(); // Default empty Value
+    Location biasLoc =
+        hasBias ? firstConv.getB().getLoc()
+                : loc; // Get location from the first Conv bias if available
+    if (hasBias) {
+      for (auto conv : convOps)
+        biasTensors.push_back(conv.getB());
+
+      Type newBiasType =
+          RankedTensorType::get({total_C_out}, weightType.getElementType());
+      axis0 = IntegerAttr::get(si64Type, 0);
+
+      concatenatedBias =
+          rewriter.create<ONNXConcatOp>(biasLoc, newBiasType, biasTensors,
+              axis0); // Bias should be concatenated along axis=0
+    }
+
+    // **Create new Grouped ConvOp**
+    auto newConv = rewriter.create<ONNXConvOp>(loc, resultType, input,
+        concatenatedWeight, hasBias ? concatenatedBias : Value(), autoPadAttr,
+        dilationsAttr, groupAttrVal, kernelShapeAttr, padsAttr, stridesAttr);
```

Comment:
Please use DialectBuilder for conv, c.f. https://github.com/onnx/onnx-mlir/blob/main/src/Dialect/ONNX/DialectBuilder.hpp#L76